src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -215,38 +294,68 @@ void Write<T>([AllowNull] T value, NpgsqlParameter param)
             param.ResolveHandler(_connector.TypeMapper);
             param.ValidateAndGetLength();
             param.LengthCache?.Rewind();
-            param.WriteWithLength(_buf, false);
+            await param.WriteWithLength(_buf, async);
             param.LengthCache?.Clear();
             _column++;
         }
 
         /// <summary>
         /// Writes a single null column value.
         /// </summary>
-        public void WriteNull()
+        public void WriteNull() => WriteNull(false).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Writes a single null column value.
+        /// </summary>
+        public Task WriteNullAsync(CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return Task.FromCanceled(cancellationToken);
+            using (NoSynchronizationContextScope.Enter())
+                return WriteNull(true);
+        }
+
+        async Task WriteNull(bool async)
         {
             CheckReady();
             if (_column == -1)
                 throw new InvalidOperationException("A row hasn't been started");
 
             if (_buf.WriteSpaceLeft < 4)
-                _buf.Flush();
+                await _buf.Flush(async);
 
             _buf.WriteInt32(-1);
             _column++;
         }
 
         /// <summary>
         /// Writes an entire row of columns.
-        /// Equivalent to calling <see cref="StartRow"/>, followed by multiple <see cref="Write{T}(T)"/>
+        /// Equivalent to calling <see cref="StartRow()"/>, followed by multiple <see cref="Write{T}(T)"/>
```

Comment:
Any reason for the parentheses after `StartRow`? Seems like they can be removed.

Reply:
Without it we get 
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0419	Ambiguous reference in cref attribute: 'StartRow'. Assuming 'NpgsqlBinaryImporter.StartRow()', but could have also matched other overloads including 'NpgsqlBinaryImporter.StartRow(bool)'.	Npgsql (net461)	D:\work\projects\npgsql\src\Npgsql\NpgsqlBinaryImporter.cs	341	Active
```

Reply:
@roji, if there are multiple overloads, then you must specify parameters of the required one.

Reply:
OK, didn't get any warning here but who cares